### PR TITLE
[simd/v3i]: Implement v128 integer basic arithmetic instructions

### PR DESCRIPTION
**Pull request by haoyu-zc**

### CODE REVIEW

**Comment by haoyu-zc** [comment]
Tested by:
```
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i64x2_arith.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i32x4_arith.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i16x8_arith.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i8x16_arith.bin.wast
```

File: src/engine/V3Eval.v3 [hunk 1]
```diff
@@ -379,6 +424,52 @@ component V3Eval {
...
+		var r2 = f(u16.view(a.0 >> 32), u16.view(b.0 >> 32));
+		var r3 = f(u16.view(a.0 >> 48), u16.view(b.0 >> 48));
+		var r4 = f(u16.view(a.1), u16.view(b.1));
+		var r5 = f(u16.view(a.1 >> 16), u16.view(b.1 >> 16));
+		var r6 = f(u16.view(a.1 >> 32), u16.view(b.1 >> 32));
+		var r7 = f(u16.view(a.1 >> 48), u16.view(b.1 >> 48));
+		return ((u64.view(r3) << 48) | (u64.view(r2) << 32) | (u64.view(r1) << 16) | r0,
+			(u64.view(r7) << 48) | (u64.view(r6) << 32) | (u64.view(r5) << 16) | r4);
+	}
+	private def do_vv_v_x16(a: (u64, u64), b: (u64, u64), f: (u8, u8) -> u8) -> (u64, u64) { // Performs a 16-lane binop
```
Comment (titzer) [inline src/engine/V3Eval.v3:446]: It's probably more efficient to just do a loop for this one.

**Comment by haoyu-zc** [comment]
Refactored the code to use loops for computations with >8 lanes

File: src/engine/V3Eval.v3 [hunk 2]
```diff
@@ -379,6 +424,53 @@ component V3Eval {
...
+		return (result0, result1);
+	}
+	private def do_vv_v_x16(a: (u64, u64), b: (u64, u64), f: (u8, u8) -> u8) -> (u64, u64) { // Performs a 16-lane binop
+		var result0: u64 = 0;
+		var result1: u64 = 0;
+
+		for (i < 8) {
+			var shift_amount: byte = byte.!(i * 8);
+			
+			var r_a = u8.view((a.0 >> shift_amount) & 0xFF);
```
Comment (titzer) [inline src/engine/V3Eval.v3:461]: You can drop the `0xFF`, as that is redundant with `u8.view`.

File: src/engine/V3Eval.v3 [hunk 3]
```diff
@@ -379,6 +424,53 @@ component V3Eval {
...
+			result1 |= (u64.view(res) << shift_amount);
+		}
+
+		return (result0, result1);
+	}
+	private def do_vv_v_x16(a: (u64, u64), b: (u64, u64), f: (u8, u8) -> u8) -> (u64, u64) { // Performs a 16-lane binop
+		var result0: u64 = 0;
+		var result1: u64 = 0;
+
+		for (i < 8) {
```
Comment (titzer) [inline src/engine/V3Eval.v3:458]: You can do `for (shift: byte = 0; shift < 64; shift += 8) {`

Comment (haoyu-zc) [reply src/engine/V3Eval.v3]: Right right. Got rid of that i and renamed some locals.

File: src/engine/V3Eval.v3 (hunk 2 above)
Comment (haoyu-zc) [reply src/engine/V3Eval.v3]: Right right. Got rid of that i and renamed some locals

**Comment by haoyu-zc** [comment]
Removed unnecessary locals. Renamed result0 and result1 to low and high

**Review by titzer — APPROVED** [review]
lgtm

**Pull request merged**